addons/functions/functions/fnc_removeTrench.sqf

```diff
@@ -118,28 +96,25 @@ private _fnc_condition = {
         [_handle] call CBA_fnc_removePerFrameHandler;
     };
 
-    private _animationPhase = _trench animationSourcePhase "rise";
-    private _diff = (_trench getVariable [QGVAR(diggingSteps), (([configFile >> "CfgVehicles" >> typeOf _trench >> QGVAR(offset), "NUMBER", 2] call CBA_fnc_getConfigEntry)/(_removeTime*10))]) * _diggerCount;
- 
-    _trench animateSource ["rise", _animationPhase - _diff, true];
-    _trench setVectorDirAndUp _vecDirAndUp;
+    private _diggerCount = count (_trench getVariable [QGVAR(diggers),[]]);
+    private _newProgress =  _actualProgress - ((1/_removeTime)/10) * _diggerCount;
+    _trench setVariable ["ace_trenches_progress", _newProgress, true];
```

Comment:
As far as I can see, `ace_trenches_progress` it is only ever set in conjunction with calling `FUNC(setTrenchProgress)`, so it can be moved into the function.